src/Microsoft.Health.Fhir.Core/Features/Operations/Export/ExportProcessingJob.cs

```diff
@@ -94,9 +95,16 @@ record = exportJobRecord;
                     // For single threaded jobs this section will complete a job after every page of results is processed and make a new job for the next page.
                     // This will allow for saving intermediate states of the job.
 
-                    var definition = jobInfo.DeserializeDefinition<ExportJobRecord>();
-                    definition.Progress = exportJobRecord.Progress;
-                    await _queueClient.EnqueueAsync(QueueType.Export, innerCancellationToken, jobInfo.GroupId, definitions: definition);
+                    record.Status = OperationStatus.Completed;
+                    var existingJobs = await _queueClient.GetJobByGroupIdAsync(QueueType.Export, jobInfo.GroupId, false, cancellationToken);
+
+                    // Checks that a followup job has not already been made.
+                    if (!existingJobs.Any((job) => job.Id > jobInfo.Id))
```

Comment:
qq, why was the new definition always different when we requeued?

https://github.com/microsoft/fhir-server/blob/90c537ee52e64cd041bb2354de2374d747f68844/src/Microsoft.Health.Fhir.CosmosDb/Features/Storage/Queues/CosmosQueueClient.cs#L66-L77

Reply:
My guess is because of the CT included in the progress. I haven't paid attention to Cosmos CTs to see if they change over time.